### PR TITLE
[Merged by Bors] - fix: patch for std4#198 (more mul lemmas for `Nat`)

### PR DESCRIPTION
**Pull request by fgdorais**
---

- [x] depends on: https://github.com/leanprover/std4/pull/198

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/EuclideanDomain/Instances.lean [hunk 1]
```diff
@@ -35,7 +35,7 @@ instance Int.euclideanDomain : EuclideanDomain ℤ :=
       not_lt_of_ge <| by
         rw [← mul_one a.natAbs, Int.natAbs_mul]
         rw [←Int.natAbs_pos] at b0
-        exact Nat.mul_le_mul_of_nonneg_left b0 }
```
Comment (eric-wieser) [inline Mathlib/Algebra/EuclideanDomain/Instances.lean:38]: FWIW, this old (and incorrect) name is left from https://github.com/leanprover-community/lean/pull/366

Comment (fgdorais) [reply Mathlib/Algebra/EuclideanDomain/Instances.lean:38]: At least the spurious assumption was eliminated!

**Comment by fgdorais** [comment]
I'm stumped on the last error in `Mathlib.FieldTheory.SeparableDegree`, any thoughts? @eric-wieser @semorrison

**Comment by semorrison** [comment]
Unfortunately this branch doesn't have oleans, and I don't want to take the time to build myself right now. Could you merge and ping again?

**Comment by semorrison** [comment]
bors p=10

**Pull request closed without merging**